### PR TITLE
roachprod: fix unused `serviceMode` parameter

### PR DESCRIPTION
**Pull request by herkolategan**
Previously, `servicesWithOpenPortSelection` did not use the `serviceMode` parameter that was passed, and it assumed `ServiceModeExternal` in the function itself.

This was updated to use the `serviceMode` parameter instead, but caused issues during start-up, because the mode being passed in for the storage cluster was shared instead of external.

The mode has been updated to `ServiceModeExternal` now for storage clusters (`StartDefault` with System Interface). A comment has also been added to avoid confusion in the future regarding if the storage cluster is shared or external (it's external).

In conclusion, only shared process tenants (non-system) are regarded shared. Any other registered services are regarded as external.

Epic: None
Release Note: None

### CODE REVIEW

**Review by DarrylWong — APPROVED** [review]

**Comment by herkolategan** [comment]
TFTR!

bors r=DarrylWong
